CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -97,8 +97,21 @@ function Staff:tickDay()
    self:changeAttribute("happiness", 0.05)
   end
 
-  --TODO windows in your work space and a large space to work in add to happiness
-  -- working in a small space makes you unhappy
+  local room = self:getRoom()
+  -- It always makes you happy to see the outdoors (or windows to anywhere)
+  if room then
+    local count = room:countWindows()
+    self:changeAttribute("happiness", 0.001 * count)
+    if room.room_info.id == "staff_room" then -- Pleased another bit
+      self:changeAttribute("happiness", 0.001 * count)
+    end
+  end
+
+  -- Extra space in the room you are in adds to your happiness
+  if room and room.minimum_size and room.width > room.minimum_size and
```

Comment:
Only extra space in both dimensions no matter how much?

Reply:
It could be the amount of extra space in relation to minimum, say factor = (width * height) / (minimum_size * miminum size) / 1000. Could be log scale, to keep a 20x20 staff room from adding 0.025.